core/arch/arm/kernel/thread.c

```diff
@@ -358,7 +358,8 @@ static void init_regs(struct thread_ctx *thread,
 	 * Stdcalls starts in SVC mode with masked IRQ, masked Asynchronous
 	 * abort and unmasked FIQ.
 	  */
-	thread->regs.cpsr = CPSR_MODE_SVC | CPSR_I | CPSR_A;
+	thread->regs.cpsr = read_cpsr();
```

Comment:
Please mask out the bit that should be preserved from CPSR instead of taking all.